specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json

```diff
@@ -0,0 +1,1597 @@
+          }
+        }
+      },
+      "description": "An object that represents the approval state of the private link connection.",
+      "type": "object"
+    },
+    "MHSMPrivateEndpointServiceConnectionStatus": {
+      "type": "string",
+      "description": "The private endpoint connection status.",
+      "enum": [
+        "Pending",
+        "Approved",
+        "Rejected",
+        "Disconnected"
+      ],
+      "x-ms-enum": {
+        "name": "PrivateEndpointServiceConnectionStatus",
+        "modelAsString": true
+      }
+    },
+    "MHSMPrivateEndpointConnectionProvisioningState": {
+      "type": "string",
+      "readOnly": true,
+      "description": "The current provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Creating",
+        "Updating",
+        "Deleting",
+        "Failed",
+        "Disconnected"
+      ],
+      "x-ms-enum": {
+        "name": "PrivateEndpointConnectionProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "MHSMPrivateLinkResourceListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "Array of private link resources",
+          "items": {
+            "$ref": "#/definitions/MHSMPrivateLinkResource"
+          }
+        }
+      },
+      "description": "A list of private link resources",
+      "type": "object"
+    },
+    "MHSMPrivateLinkResource": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/MHSMPrivateLinkResourceProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ManagedHsmResource"
+        }
+      ],
+      "description": "A private link resource",
+      "type": "object"
+    },
+    "MHSMPrivateLinkResourceProperties": {
+      "properties": {
+        "groupId": {
+          "description": "Group identifier of private link resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "requiredMembers": {
+          "description": "Required member names of private link resource.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "requiredZoneNames": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Required DNS zone names of the the private link resource."
+        }
+      },
+      "description": "Properties of a private link resource.",
+      "type": "object"
+    },
+    "CheckMhsmNameAvailabilityParameters": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The managed hsm name."
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "The parameters used to check the availability of the managed hsm name."
+    },
```

Comment:
keyvault defined the payload slightly different.
https://github.com/Azure/azure-rest-api-specs/blob/ff63535af8d98b47d1a57aa9796cba4ae070e393/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/keyvault.json#L1685-L1709

I don't think there is requirement from ARM on the schema, so it is up to you.

Reply:
@neeerajaakula or @riteshchitlangia1 : Please confirm that the slight discrepancies here are acceptable for Managed HSM.

Reply:
This is not relevant for Managed HSM api